Document setup cases (#703)
software-mansion/protostar: website/docs/tutorials/07-testing/README.md

You can find all [assert signatures here](https://github.com/software-mansion/protostar/blob/master/cairo/protostar/asserts.cairo).

## Setup hooks

Often while writing tests you have some setup work that needs to happen before tests run.

> The `__setup__` and `setup_*` hooks can simplify and speed up your tests.

Reviewer: ```suggestion
The `__setup__` and `setup_<test_name>` hooks can simplify and speed up your tests.
```
It suggests glob

> 

Reviewer: ```suggestion

#### Global setup hook
```
If we give case hook a name. I would name the global one for the sake of consistency.

Author: Done.

> 1. The `__setup__` hook is shared between all test cases in module, and is executed before

Reviewer: ```suggestion
1. The `__setup__` hook is shared between all test cases in a module, and is executed before
```

   test case.

> 2. The `setup_*` case hook is bound to a matching `test_*` case and is executed between

Reviewer: ```suggestion
#### Case setup hook
2. The `setup_*` case hooks are bound to a matching `test_*` cases and are executed between
```

Author: Done.

>    the `__setup__` hook and the test case.

Reviewer: ```suggestion
   the global setup hook and the test case.
```

Author: Done.

end
```

```cairo title="Using setup_* case hook"
%lang starknet

@external
func setup_test_something():

>     %{ max_examples(500) %}

Reviewer: Maybe `deploy_contract` instead of `max_examples` would make this example simpler?
```suggestion
    %{ max_examples(500) %}
```

Author: `deploy_contract` does make a lot of sense within setup case, you can put it inside test case directly and this will work.

`max_examples` instead is `setup` specific and would often be used, hence this example.

Reviewer: Yea, deploy contract is a bit weird. I just have a feeling that `max_examples` is not a simple enough cheatcode for an example, as you have to know what fuzzing is beforehand which makes learning curve steeper. 

Alternative idea, which _maybe_ tells a better story
```
    func __setup__():
        deploy_contract()
    end

     func setup_test_a():
        store(5)
    end

     func test_a():
        assert load() = 5
    end

     func test_b():
        assert load() = 0
    end
```

leaving the decision to you

Author: #726 I very very like your idea for an example. Let's do this later in separate PR.
